GitHubConfiguration.ps1

```diff
@@ -197,7 +208,9 @@ function Set-GitHubConfiguration
         [ValidateRange(0, 3600)]
         [int] $WebRequestTimeoutSec,
 
-        [switch] $SessionOnly
+        [switch] $SessionOnly,
+
+        [string] $GitHubBaseUrl
```

Comment:
Would be worthwhile to help prevent users from shooting themselves in the foot by validating what they're passing in and ensuring that they're not including `http://`, `https://`, `www.` or `api.` as part of their BaseUrl:

```
   [ValidatePattern('^(?!https?)(?!api\.)(?!www\.).*')]
```

Reply:
Implemented with a minor addition, I've added a colon to the end of the first negative lookahead so that `https://git.contoso.com` would be caught while `httpgithub.contoso.com` would be fine. 

This uh, this one comes from experience with an _interesting_ company. 🙃